misc/systemd/ipfs@.service

```diff
@@ -0,0 +1,31 @@
+[Unit]
```

Comment:
Users will likely need to just write their own custom unit in this case. Providing a templated unit doesn't really buy much.

Reply:
Well, the [archlinux package](https://git.archlinux.org/svntogit/community.git/tree/trunk/ipfs@.service?h=packages/go-ipfs) provides such a ipfs@.service file, because it's much simpler for the users to just add an override config with `systemctl edit ipfs@user` than to create a custom service file.

This way the user could add just the necessary changes, while everything else gets updated by the package.

The idea was to remove the necessity to provide a custom service file by bringing it upstream. I don't think it hurts to provide this file, does it?

Reply:
Ideally, packages would use a dynamic user or a special `ipfs` user.

If users just want to run an IPFS daemon as a specific user, they should:

1. Enable `user@$uid.service` on start.
2. Enable the `ipfs` unit for that user.

Reply:
I agree that users should use a user-daemon for most situations - this is already documented [in the ArchLinux wiki](https://wiki.archlinux.org/index.php/IPFS#Using_a_service_to_start_the_daemon).

So what's now the sane default? Change the regular `ipfs.service` to run as user `ipfs` and advice the package maintainer that they have to create system-user `ipfs`?

Because running ipfs as root isn't a great default.

Reply:
> So what's now the sane default? Change the regular ipfs.service to run as user ipfs and advice the package maintainer that they have to create system-user ipfs?

SGTM. But if we do that, we should change the default home to /var/lib/ipfs

Reply:
In the service file as an environment parameter?

Reply:
I've added a sysusers.conf with the path and set the working directory accordingly.